### PR TITLE
add rate limit

### PR DESCRIPTION
**Pull request by mattops**

### CODE REVIEW

File: helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml [hunk 1]
```diff
@@ -12,6 +12,11 @@ metadata:
     heritage: {{ .Release.Service }}
   annotations:
     kubernetes.io/ingress.class: "nginx"
+    # note this limit is per ingress replica
+    nginx.ingress.kubernetes.io/limit-rpm: 10
```
Comment (mikehalmamoj) [inline helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml:16]: 10 per minute?  Seems a bit tight.

File: helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml [hunk 2]
```diff
@@ -12,6 +12,11 @@ metadata:
...
+    # setting to 429 (default is 503) so we can alert on this
```
Comment (mikehalmamoj) [inline helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml:17]: default is 503 :rofl: If only there was an http status code called something like 429=Too Many Requests.

File: helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml (hunk 1 above)
Comment (mattops) [reply helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml:16]: yes initially I thought that too - but remember it's per ingress instance.  Each request round-robins over many ingress instances.

Comment (mattops) [reply helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml:16]: Also there is a default burst value of 10x that number.

Comment (mikehalmamoj) [reply helm_deploy/manage-hmpps-auth-accounts/templates/ingress.yaml:16]: Fair enough. I imagine there will be some fine tuning required whatever you pick.

**Pull request merged**